### ISSUE TITLE
Support for ARM?

### ISSUE THREAD

**Issue by saimon-moore** [thayakawa-gh/SlackLogViewer#10]
Hi,

I just tried this on an M1 and it crashes with a segfault just as soon as I try to open a folder exported with https://github.com/rusq/slackdump/blob/master/doc/usage-export.rst#export-types (standard export)

<img width="1728" alt="Screenshot 2022-11-29 at 09 48 47" src="https://user-images.githubusercontent.com/104002291/204482250-67e2b21d-58b8-4b8b-9782-0215713be3dd.png">

<img width="1728" alt="Screenshot 2022-11-29 at 09 48 22" src="https://user-images.githubusercontent.com/104002291/204482223-d613eee9-b27b-46a8-900e-e30fa67d355a.png">

Any chance, a build for arm will become available soon?

**Comment by thayakawa-gh** [comment]
Hi, @saimon-moore. Sorry for the inconvenience.
Because I will be very busy for the next few months, I can't release it right away.
In SlackLogViewer, Qt5 is used as a GUI toolkit, but it doesn't seem to support Apple Silicon natively, so I have to upgrade the Qt version to 6. In addition, I don't have a mac so I can't easily build/test. I need someone's help to release the binary.
From the above, I don't know when I will be able to work on it. I hope you will wait for a while until I can make time to do so.

**Comment by saimon-moore** [comment]
Yes I will wait patiently. Unfortunately I don't have the chops to try it myself :(

Thank you very much

**Comment by norim13** [comment]
Hey 👋 I haven't run any C code since college, so my knowledge is pretty much gone, but if someone can give me some guidance I can help try to build this on Apple silicon.

(getting the segmentation fault myself too)

**Comment by thayakawa-gh** [comment]
Hi, saimon-moore and norim13. Sorry for my late reply.
Now I've got some time and started working on migration of SlackLogViewer to Qt6.
I'll provide you a pilot version soon, so it would be helpful if you could test it on Apple Silicon.

**Comment by thayakawa-gh** [comment]
Now test builds (using [Qt5](https://github.com/thayakawa-gh/SlackLogViewer/actions/runs/4713797463) or [Qt6](https://github.com/thayakawa-gh/SlackLogViewer/actions/runs/4713797462)) for macOS are available. They can be downloaded from the links -> Artifacts -> macos or macos-qt6.
Could you try them on your mac?

**Comment by norim13** [comment]
Just tried both, but unfortunately I think none worked :\ It crashes immediately once I select the folder I want to open. Is there any way for me to give you better data about the actual error? Any logs I can fetch somewhere?

**Comment by thayakawa-gh** [comment]
Hmm, the cause of the crash should be isolated first.

1. Apple silicon
We need someone to test the latest SlackLogViewer on both apple silicon and intel cpu.
2. Problems with the export file formatting
If you have export files from various workspaces, it would be helpful if you could test as many of them as possible.
3. Simple bugs in SlackLogViewer
I think this is the most likely scenario. I'm working on adding a function to display error messages when critical problems occur.

In fact, it would be great if someone could debug it on mac...

**Comment by sclsj** [comment]
Since this issue hasn't been closed, I assume you would still need someone to test on Apple Silicon? I have a m1 max macbook and is glad to help.

**Comment by sclsj** [comment]
Also, have you considered releasing two versions (two .tar.xz files or two .apps in one .tar.xz file in this case), one for each architecture? Personally I generally dislike Universal Binaries because it wastes space while adding no real benefit to me, and it seems to complicate the build process sometimes, I heard.

**Comment by thayakawa-gh** [comment]
Hello, @sclsj. Sorry for my late reply.
Since I'm not familiar with the build process on macOS, I have just followed Qt instructions, etc. and the build commands are described in `.github/workflows/macos-qt6.yaml`. I'm not sure what the difference is between the universal binaries and individual .apps.
Is there a possibility that separating .apps could solve the issues mentioned in this thread? If so, I will look into it further.

Off course, anyone willing to test is welcome.
